app/controllers/sessions_controller.rb

```diff
@@ -52,4 +53,15 @@ def set_csrf_headers
     response.headers['X-CSRF-Param'] = request_forgery_protection_token.to_s
     response.headers['X-CSRF-Token'] = form_authenticity_token.to_s
   end
+
+  def revoke_access_tokens!
+    application_id_to_revoke = doorkeeper_token&.application_id
+    return unless application_id_to_revoke
+
+    user = User.find(doorkeeper_token&.resource_owner_id)
```

Comment:
more safe operators! these should be done once via the guard clause / conditional before the method is called.